### ISSUE TITLE
HTML Validation Issues

### ISSUE THREAD

**Issue by kraftbj** [MichaelArestad/Press-This#57]
Per https://make.wordpress.org/core/2015/02/11/press-this-update-210/ , HTML validation needs to be examined before next week. I ran a test site's press-this.php (source of a directly-accessed page, no bookmarklet usage, did not scrape a page) through W3C's validator:

- [x] 1. Warning: The language attribute on the script element is obsolete. You can safely omit it. ( https://github.com/MichaelArestad/Press-This/blob/436d0d8cc1ee824c08680eb6baaf1c0968833a7c/press-this.php#L637 )
- [x] 2. Error: Bad value for attribute action on element form: Must be non-empty. ( https://github.com/MichaelArestad/Press-This/blob/436d0d8cc1ee824c08680eb6baaf1c0968833a7c/press-this.php#L697 )
- [x] 3. Error: Bad value for attribute action on element form: Must be non-empty. ( https://github.com/MichaelArestad/Press-This/blob/436d0d8cc1ee824c08680eb6baaf1c0968833a7c/press-this.php#L703 )
- [ ] 4. Warning:  Empty heading. ( https://github.com/MichaelArestad/Press-This/blob/436d0d8cc1ee824c08680eb6baaf1c0968833a7c/press-this.php#L712 )
- [x] 5. Error: Element link is missing required attribute property.  (Not sure the line item, think it is from TinyMCE and may be a Core issue: `<div id="wp-pressthis-wrap" class="wp-core-ui wp-editor-wrap tmce-active"><link rel='stylesheet' id='editor-buttons-css' href='http://kraftbj.wpsandbox.me/wp-includes/css/editor.min.css?ver=4.2-alpha-31428' type='text/css' media='all' />` )
- [x] 5b. Error: Attribute autocomplete not allowed on element textarea at this point. (Same here, next line in the final output: `<textarea class="wp-editor-area" style="height: 350px" autocomplete="off" cols="40" name="pressthis" id="pressthis">` )
- [x] 6. Error: The element label must not appear as a descendant of the a element. ( https://github.com/MichaelArestad/Press-This/blob/436d0d8cc1ee824c08680eb6baaf1c0968833a7c/press-this.php#L749 ) 
- [x] 7. Error: The element label must not appear as a descendant of the a element. ( https://github.com/MichaelArestad/Press-This/blob/436d0d8cc1ee824c08680eb6baaf1c0968833a7c/press-this.php#L756 )
- [x] 8. Error: The element label must not appear as a descendant of the a element. ( https://github.com/MichaelArestad/Press-This/blob/436d0d8cc1ee824c08680eb6baaf1c0968833a7c/press-this.php#L762 )
- [x] 9. Error: The for attribute of the label element must refer to a form control. ( https://github.com/MichaelArestad/Press-This/blob/436d0d8cc1ee824c08680eb6baaf1c0968833a7c/press-this.php#L711 )

**Comment by DrewAPicture** [comment]
RE: the empty `action` attributes, see https://core.trac.wordpress.org/changeset/31200 for the core solution to this.

tl;dr If the action attributes are empty, just remove them

**Comment by kraftbj** [comment]
For the label items (6-9), basically, labels are only for form elements. Is there another method of the same thing that is valid? `title` as an attribute seems close, though most things say to avoid using it.

**Comment by marcelomazza** [comment]
We can use `aria-labelledby` and replace the `labels` with other elements.

**Comment by kraftbj** [comment]
@marcelomazza :+1: Can I leave items (6-9) for you?

Items 5 and 6 occur on post-new.php and are inherited from Core/TinyMCE. Marking them as resolved for our end to defer up to Core.

**Comment by marcelomazza** [comment]
@kraftbj yeap! resolved (6-9) in PR: https://github.com/MichaelArestad/Press-This/pull/60

**Comment by MichaelArestad** [comment]
Ideas on #4? Feels weird to put content there just to validate.

**Comment by kraftbj** [comment]
Well, it's only a warning. What's our thoughts on letting a warning stand? Going to revalidate and open core tickets for the TinyMCE fails.

**Comment by marcelomazza** [comment]
@MichaelArestad kind of ugly but it may work: we can put the same "Title" text that's on the `label` and then overlay both strings one on front of the another... it's the only thing I can think of. And if we do that we need to add JS placeholder logic for the `h2`too.

**Comment by MichaelArestad** [comment]
@marcelomazza Let's not do that.

**Comment by kraftbj** [comment]
For the TinyMCE issues: https://core.trac.wordpress.org/ticket/31321#ticket

Once I remove the offending TinyMCE entries, we validate as of 440ffdf , so I am leaning toward ignoring the warning if @DrewAPicture :+1: it.

Suggestion: Leave this open as a reminder to double-check once the code is fully stable, but no further action at this time.

**Comment by MichaelArestad** [comment]
@DrewAPicture Any word on #4?

**Comment by DrewAPicture** [comment]
@MichaelArestad IIRC, there were accessibility concerns. I'd suggest maybe hitting up @afercia.

**Comment by kraftbj** [comment]
The latest commit passes validation, with the core TinyMCE issues ignored. Closing the ticket.
